src/app/custom-navigation-mode/custom-navigation-mode.ts

```diff
@@ -0,0 +1,36 @@
+import {SemiStrictNavigationMode} from 'angular-archwizard';
+
+/**
+ * A custom NavigationMode
+ *
+ * Allows to navigate backward to any step (like [[StrictNavigationMode]])
+ * and forward to steps the user has already visited before.
+ *
+ * Next and Previous buttons work as in [[StrictNavigationMode]] and [[SemiStrictNavigationMode]].
+ */
+export class CustomNavigationMode extends SemiStrictNavigationMode {
+
+  // @override
+  protected canTransitionToStep(destinationIndex: number) {
+
+    // Use the base implementation of `isNavigable` from `SemiStrictNavigationMode`
+    return super.isNavigable(destinationIndex);
+  }
+
+  // @override
+  public isNavigable(destinationIndex: number): boolean {
+
+    // Allow returning to previous steps
+    if (destinationIndex < this.wizardState.currentStepIndex) {
+      return true;
+    }
+
+    // Allow returning to any previously visited ("completed" steps)
+    const step = this.wizardState.wizardSteps[destinationIndex];

```

Comment:
I think we should use `this.wizardState.getStepAtIndex(destinationIndex)` here instead of the direct array access.
The reason for using the getter method is to abstract from the underlying implementation details of the `WizardState` class. In theory this makes it easier to change the internals of the `WizardState` class without touching other components, if for example we want to change the array to a different component.

Reply:
OK. Here `wizardSteps` are used directly because that's how it is done in built-in navigation modes.

Reply:
Changed to `getStepAtIndex` in 049d50d6.

Reply:
Ok, I didn't notice that we used direct array access in the built-in navigation modes too. I think we should change it there too.

Reply:
I was mistaken. Built-in navigation modes correctly use `getStepAtIndex`. They access `wizardSteps` directly only for things not exposed through methods (like `wizardSteps.length` or `wizardSteps.filter(...)`). I think there is nothing to fix there now.